packages/block-editor/src/hooks/position.js

```diff
@@ -222,32 +279,39 @@ export function PositionPanel( props ) {
 	const allowSticky = hasStickyPositionSupport( blockName );
 	const value = style?.position?.type;
 
-	const { hasParents } = useSelect(
+	const { firstParentClientId } = useSelect(
 		( select ) => {
 			const { getBlockParents } = select( blockEditorStore );
 			const parents = getBlockParents( clientId );
-			return {
-				hasParents: parents.length,
-			};
+			return { firstParentClientId: parents[ parents.length - 1 ] };
 		},
 		[ clientId ]
 	);
 
+	const blockInformation = useBlockDisplayInformation( firstParentClientId );
```

Comment:
Oh, that is a clever idea — I can't remember why exactly, but I think from memory there was a preference to try to use popovers so that we don't accidentally collide with any styles that a theme might have for a particular block. It could be to ensure that the visualizers / popover always displays its CSS styling in addition to whatever might be set on the parent block (that we can't quite know) rather than overriding anything. Worth playing with though!

Reply:
I picked outline because it's very unlikely that it gets used for block styling, as it's mostly a property applied on focus. Though if the popover calculations are performant I suppose it's not really an issue.